join.go

```diff
@@ -28,23 +28,13 @@ var ErrSymlinkLoop = errors.Wrap(syscall.ELOOP, "secure join")
 // accessed does not exist (or path components don't exist). This is
 // effectively a more broad version of os.IsNotExist.
 func IsNotExist(err error) bool {
-	// If it's a bone-fide ENOENT just bail.
-	if os.IsNotExist(errors.Cause(err)) {
```

Comment:
I'm not sure what kind of errors are supposed to be supplied as an argument here, but in case the error was obtained using `errors.Wrap()` from `pkg/errors` version > 0.9.0, the new code won't work as the `Unwrap` method only appeared in pkg/errors 0.9.0.

So, this either needs to be documented. Alternatively, we can do a runtime check, something like "if this is an error which has `Cause` method, it should also have `Unwrap` method, otherwise paniс saying the caller is using old pkg/errors". Complicated...

Reply:
Ah, good one; I wanted to answer "but go mod enforces pkg/errors 0.9.1, but we remove that now of course 🤔

We can do a quick check which projects use this projects as a dependency (I think that list is limited, so we can at least make sure that they use a current pkg/errors)

Reply:
Doing a quick scan using https://grep.app/search?q=github.com/cyphar/filepath-securejoin


Consumers (using pkg/errors v0.9.1))

- https://github.com/bcicen/ctop
- https://github.com/cloudfoundry/cli
- https://github.com/containers/podman
- https://github.com/cri-o/cri-o
- https://github.com/genuinetools/img
- https://github.com/google/cadvisor
- https://github.com/hashicorp/nomad
- https://github.com/helm/helm
- https://github.com/kubernetes/autoscaler
- https://github.com/kubernetes/kubernetes
- https://github.com/kubesphere/kubesphere
- https://github.com/moby/moby
- https://github.com/opencontainers/runc
- https://github.com/opencontainers/umoci
- https://github.com/openshift/origin
- https://github.com/lunarway/release-manager
- https://github.com/goharbor/harbor
- https://github.com/rancher/k3s
- https://github.com/elastic/beats
- https://github.com/kubeedge/kubeedge
- https://github.com/istio/istio
- https://github.com/kubeapps/kubeapps
- https://github.com/hpcng/singularity
- https://github.com/kubernetes/minikube
- https://github.com/datawire/ambassador
- https://github.com/rancher/rancher

Indirect

- https://github.com/linkerd/linkerd2 (indirect)
- https://github.com/databus23/helm-diff (indirect)
- https://github.com/derailed/k9s (indirect)
- https://github.com/operator-framework/operator-sdk (indirect)

Indirect / potentially affected

- https://github.com/fluxcd/flux (old pkg/errors, but using go 1.15)
- https://github.com/flynn/flynn (old pkg/errors)
- https://github.com/bblfsh/bblfshd (indirect dependency, uses older pkg/errors)
- https://github.com/ContainerSolutions/helm-monitor (indirect dependency, uses older pkg/errors)
- https://github.com/deepmind/kapitan (indirect dependency, uses older pkg/errors)
- https://github.com/helm/monocular (indirect dependency, uses older pkg/errors)